apps/frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,20 @@
+import { Navigate, Outlet } from "react-router";
+
+import { AppRoute } from "~/libs/enums/app-route.enum.js";
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+
+const ProtectedRoute = (): React.JSX.Element => {
+	const authenticatedUser = useAppSelector(
+		({ auth }) => auth.authenticatedUser,
+	);
+
+	const isAuthenticated = Boolean(authenticatedUser);
+
+	return isAuthenticated ? (
+		<Outlet />
+	) : (
+		<Navigate replace to={AppRoute.SIGN_IN} />
+	);
+};
```

Comment:
Instead of the outlet, this route needs to accept any children (page), await the user's status while showing a loader, and then either redirect to sign-in or return the page
```suggestion
const ProtectedRoute = ({ children }: Properties): React.JSX.Element => {
	const authenticatedUser = useAppSelector(
		({ auth }) => auth.authenticatedUser,
	);
	const authDataStatus = useAppSelector(
		({ auth }) => auth.dataStatus,
	);

	const isLoading =
		dataStatus === DataStatus.PENDING || dataStatus === DataStatus.IDLE;

	if (isLoading) {
		return (
			<div className={styles["loader-container"]}>
				<Loader />
			</div>
		);
	}
	
	if (!authenticatedUser) {
		return <Navigate replace to={AppRoute.SIGN_IN} />;
	}

	return <>{children}</>;
};
```